### PR TITLE
Add a Java API for response headers

### LINKED ISSUE #2178: Improve the Play/Java Result API
Last week we had a bit of trouble working our way out through play.mvc.Result API.  brikis98 has a nice gist that demonstrates some of the pain point with action composition https://gist.github.com/brikis98/8000277

I know that Result type is supposed to be deprecated maybe there is more pending cleanup but right now its quite hard to check simple things like status of a given play.mvc.Result. There is play.core.j.JavaResultExtractor but it doesn't provide all the helper methods. Interestingly this class is only used from play.test.Helper. 

Ideally it will be great if play.mvc.SimpleResult can expose same API as Scala version.

### PR DESCRIPTION
**Pull request by marcospereira**
## Fixes

Fixes #2178.

## Purpose

Add a Java API for response headers

## References

See #2178 and related issues.

### CODE REVIEW

File: framework/src/play/src/main/java/play/mvc/Result.java [hunk 1]
```diff
@@ -130,16 +141,29 @@ public int status() {
...
-        return OptionConverters.toJava(header.reasonPhrase());
+        return header.reasonPhrase();
+    }
+
+    /**
+     * Get the Scala version of response header
+     *
+     * @return the header
+     *
+     * @deprecated Deprecated as in 2.6.0. Use {@link #getHeader()}
```
Comment (gmethvin) [inline framework/src/play/src/main/java/play/mvc/Result.java]: Shouldn't this say use `getHeader().asScala()`?

File: framework/src/play/src/main/java/play/mvc/ResponseHeader.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    public Optional<String> getHeader(String headerName) {
+        return Optional.ofNullable(this.headers.get(headerName));
+    }
+
+    public Map<String, String> headers() {
+        return Collections.unmodifiableMap(headers);
+    }
+
+    public ResponseHeader addHeader(String name, String value) {
```
Comment (gmethvin) [inline framework/src/play/src/main/java/play/mvc/ResponseHeader.java]: This should be `setHeader(name, value)`. `addHeader` suggests that the header will not replace any existing headers.  Same with `addHeaders`, though `setHeaders` could be somewhat confusing as it's not clear whether it replaces the whole set of headers or just the ones in the map.

Comment (marcospereira) [inline framework/src/play/src/main/java/play/mvc/ResponseHeader.java]: Or maybe `withHeader` and `withHeaders`? This idiom looks more consistent with other APIs.

Comment (gmethvin) [inline framework/src/play/src/main/java/play/mvc/ResponseHeader.java]: Yeah, `withHeader`/`withHeaders` sounds okay.

**Comment by marcospereira** [comment]
This is ready to be reviewed again.

**Pull request merged**